// ng-bootstrap-alert/esm2015/lib/ng-bootstrap-alert.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { NgBootstrapAlertService } from './ng-bootstrap-alert.service';
import { trigger, state, transition, style, animate } from '@angular/animations';
export class NgBootstrapAlertComponent {
    /**
     * @param {?} bootstrapAlertService
     */
    constructor(bootstrapAlertService) {
        this.bootstrapAlertService = bootstrapAlertService;
        this.DEFAULT_TIMEOUT = 3000;
        this.FADE_TIMEOUT = 600;
        this.bootstrapAlerts = [];
        this.bootstrapAlertService._bootstrapAlert.subscribe((bootstrapAlert) => {
            this.show(bootstrapAlert);
        });
    }
    /**
     * @param {?} bootstrapAlert
     * @return {?}
     */
    show(bootstrapAlert) {
        if (!bootstrapAlert) {
            return;
        }
        this.bootstrapAlerts.unshift(bootstrapAlert);
        setTimeout(() => {
            this.destroy(bootstrapAlert.uuid);
        }, bootstrapAlert.timeoutInMiliSeconds ? bootstrapAlert.timeoutInMiliSeconds : this.DEFAULT_TIMEOUT);
    }
    /**
     * @param {?} bootstrapAlert
     * @return {?}
     */
    fade(bootstrapAlert) {
        bootstrapAlert.state = 'destroyed';
    }
    /**
     * @param {?} uuid
     * @return {?}
     */
    destroy(uuid) {
        const /** @type {?} */ bootstrapAlertIndex = this.findIndex(this.bootstrapAlerts, 'uuid', uuid);
        if (bootstrapAlertIndex !== -1) {
            const /** @type {?} */ bootstrapAlert = this.bootstrapAlerts[bootstrapAlertIndex];
            bootstrapAlert.state = 'destroyed';
            setTimeout(() => {
                this.bootstrapAlerts.splice(bootstrapAlertIndex, 1);
            }, this.FADE_TIMEOUT);
        }
    }
    /**
     * @param {?} array
     * @param {?} attr
     * @param {?} value
     * @return {?}
     */
    findIndex(array, attr, value) {
        for (let /** @type {?} */ i = 0; i < array.length; i += 1) {
            if (array[i][attr] === value) {
                return i;
            }
        }
        return -1;
    }
}
NgBootstrapAlertComponent.decorators = [
    { type: Component, args: [{
                selector: 'ng-bootstrap-alert',
                template: `<div [ngClass]="alertPosition" class="bootstrap-alert col-lg-4">
	<div *ngFor="let bootstrapAlert of bootstrapAlerts" id="{{bootstrapAlert.uuid}}" class="alert alert-dismissable mb-2" [ngClass]="bootstrapAlert.type" [@alertVisible]="bootstrapAlert.state">
		<button type="button" class="close" data-dismiss="alert">x</button>
		{{bootstrapAlert.message}}
	</div>
</div>`,
                styles: [`.bootstrap-alert{z-index:2000;position:fixed}.top-left{left:1rem;top:1rem}.top-center{margin:auto;left:0;right:0;top:1rem}.top-right{right:1rem;top:1rem}.bottom-left{left:1rem;bottom:1rem}.bottom-center{margin:auto;left:0;right:0;bottom:1rem}.bottom-right{right:1rem;bottom:1rem}`],
                animations: [
                    trigger('alertVisible', [
                        state('visible', style({ opacity: 1 })),
                        state('destroyed', style({ opacity: 0 })),
                        transition('visible => destroyed', animate('600ms')),
                        transition('destroyed => visible', animate('600ms')),
                    ])
                ]
            },] },
];
/** @nocollapse */
NgBootstrapAlertComponent.ctorParameters = () => [
    { type: NgBootstrapAlertService }
];
NgBootstrapAlertComponent.propDecorators = {
    alertPosition: [{ type: Input }]
};
function NgBootstrapAlertComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    NgBootstrapAlertComponent.prototype.alertPosition;
    /** @type {?} */
    NgBootstrapAlertComponent.prototype.bootstrapAlerts;
    /** @type {?} */
    NgBootstrapAlertComponent.prototype.DEFAULT_TIMEOUT;
    /** @type {?} */
    NgBootstrapAlertComponent.prototype.FADE_TIMEOUT;
    /** @type {?} */
    NgBootstrapAlertComponent.prototype.bootstrapAlertService;
}
