// ng-bootstrap-alert/esm2015/lib/ng-bootstrap-alert.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function NgBootstrapAlertInterface() { }
function NgBootstrapAlertInterface_tsickle_Closure_declarations() {
    /** @type {?} */
    NgBootstrapAlertInterface.prototype.message;
    /** @type {?} */
    NgBootstrapAlertInterface.prototype.timeoutInMiliSeconds;
    /** @type {?} */
    NgBootstrapAlertInterface.prototype.type;
    /** @type {?} */
    NgBootstrapAlertInterface.prototype.uuid;
    /** @type {?} */
    NgBootstrapAlertInterface.prototype.state;
}
export class NgBootstrapAlert {
    /**
     * @param {?} message
     * @param {?} type
     */
    constructor(message, type) {
        this._uuid = this.generateUUId();
        this._type = type;
        this._message = message;
        this._state = 'visible';
    }
    /**
     * @return {?}
     */
    generateUUId() {
        return this.randomizer() + this.randomizer() + '-' + this.randomizer() + '-' + this.randomizer() + '-' +
            this.randomizer() + '-' + this.randomizer() + this.randomizer() + this.randomizer();
    }
    /**
     * @return {?}
     */
    randomizer() {
        return Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
    }
    /**
     * @return {?}
     */
    get message() {
        return this._message;
    }
    /**
     * @param {?} message
     * @return {?}
     */
    set message(message) {
        this._message = message;
    }
    /**
     * @return {?}
     */
    get timeoutInMiliSeconds() {
        return this._timeoutInMiliSeconds;
    }
    /**
     * @param {?} timeoutInMiliSeconds
     * @return {?}
     */
    set timeoutInMiliSeconds(timeoutInMiliSeconds) {
        this._timeoutInMiliSeconds = timeoutInMiliSeconds;
    }
    /**
     * @return {?}
     */
    get type() {
        return this._type;
    }
    /**
     * @param {?} type
     * @return {?}
     */
    set type(type) {
        this._type = type;
    }
    /**
     * @return {?}
     */
    get uuid() {
        return this._uuid;
    }
    /**
     * @return {?}
     */
    get state() {
        return this._state;
    }
    /**
     * @param {?} state
     * @return {?}
     */
    set state(state) {
        this._state = state;
    }
}
function NgBootstrapAlert_tsickle_Closure_declarations() {
    /** @type {?} */
    NgBootstrapAlert.prototype._message;
    /** @type {?} */
    NgBootstrapAlert.prototype._timeoutInMiliSeconds;
    /** @type {?} */
    NgBootstrapAlert.prototype._type;
    /** @type {?} */
    NgBootstrapAlert.prototype._uuid;
    /** @type {?} */
    NgBootstrapAlert.prototype._state;
}
