// ng-bootstrap-alert/esm5/lib/ng-bootstrap-alert.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function NgBootstrapAlertInterface() { }
function NgBootstrapAlertInterface_tsickle_Closure_declarations() {
    /** @type {?} */
    NgBootstrapAlertInterface.prototype.message;
    /** @type {?} */
    NgBootstrapAlertInterface.prototype.timeoutInMiliSeconds;
    /** @type {?} */
    NgBootstrapAlertInterface.prototype.type;
    /** @type {?} */
    NgBootstrapAlertInterface.prototype.uuid;
    /** @type {?} */
    NgBootstrapAlertInterface.prototype.state;
}
var NgBootstrapAlert = /** @class */ (function () {
    function NgBootstrapAlert(message, type) {
        this._uuid = this.generateUUId();
        this._type = type;
        this._message = message;
        this._state = 'visible';
    }
    /**
     * @return {?}
     */
    NgBootstrapAlert.prototype.generateUUId = /**
     * @return {?}
     */
    function () {
        return this.randomizer() + this.randomizer() + '-' + this.randomizer() + '-' + this.randomizer() + '-' +
            this.randomizer() + '-' + this.randomizer() + this.randomizer() + this.randomizer();
    };
    /**
     * @return {?}
     */
    NgBootstrapAlert.prototype.randomizer = /**
     * @return {?}
     */
    function () {
        return Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
    };
    Object.defineProperty(NgBootstrapAlert.prototype, "message", {
        get: /**
         * @return {?}
         */
        function () {
            return this._message;
        },
        set: /**
         * @param {?} message
         * @return {?}
         */
        function (message) {
            this._message = message;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NgBootstrapAlert.prototype, "timeoutInMiliSeconds", {
        get: /**
         * @return {?}
         */
        function () {
            return this._timeoutInMiliSeconds;
        },
        set: /**
         * @param {?} timeoutInMiliSeconds
         * @return {?}
         */
        function (timeoutInMiliSeconds) {
            this._timeoutInMiliSeconds = timeoutInMiliSeconds;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NgBootstrapAlert.prototype, "type", {
        get: /**
         * @return {?}
         */
        function () {
            return this._type;
        },
        set: /**
         * @param {?} type
         * @return {?}
         */
        function (type) {
            this._type = type;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NgBootstrapAlert.prototype, "uuid", {
        get: /**
         * @return {?}
         */
        function () {
            return this._uuid;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NgBootstrapAlert.prototype, "state", {
        get: /**
         * @return {?}
         */
        function () {
            return this._state;
        },
        set: /**
         * @param {?} state
         * @return {?}
         */
        function (state) {
            this._state = state;
        },
        enumerable: true,
        configurable: true
    });
    return NgBootstrapAlert;
}());
export { NgBootstrapAlert };
function NgBootstrapAlert_tsickle_Closure_declarations() {
    /** @type {?} */
    NgBootstrapAlert.prototype._message;
    /** @type {?} */
    NgBootstrapAlert.prototype._timeoutInMiliSeconds;
    /** @type {?} */
    NgBootstrapAlert.prototype._type;
    /** @type {?} */
    NgBootstrapAlert.prototype._uuid;
    /** @type {?} */
    NgBootstrapAlert.prototype._state;
}
